plugins/inputs/logfile/tail/tail.go

```diff
@@ -165,10 +165,9 @@ func (tail *Tail) Stop() error {
 }
 
 // StopAtEOF stops tailing as soon as the end of the file is reached.
-// Blocks until tailer is dead and returns reason for death.
-func (tail *Tail) StopAtEOF() error {
+// Does not wait until tailer is dead.
+func (tail *Tail) StopAtEOF() {
 	tail.Kill(errStopAtEOF)
-	return tail.Wait()
```

Comment:
I recommend looking over `LogFile.FindLogSrc()` to see how `Tail.StopAtEOF()` is called - https://github.com/aws/amazon-cloudwatch-agent/blob/5deb8c964efd3ac8efd5a0b27f162e7244996873/plugins/inputs/logfile/logfile.go#L174